sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/client/CosmosDb.java

```diff
@@ -74,14 +74,21 @@ public static String urlEncode(String url, String enc) {
     }
 
     public static Map<String, String> addRequiredHeaders(
-            Map<String, String> headers,
-            String partition,
-            String dbToken) {
-        headers.put("x-ms-documentdb-partitionkey", String.format("[\"%s\"]", partition));
-        headers.put("x-ms-version", "2018-06-18");
-        headers.put("x-ms-date", nowAsRFC1123());
-        headers.put("Content-Type", "application/json");
-        headers.put("Authorization", urlEncode(dbToken));
+            Map<String, String> additionalHeaders,
+            final String partition,
+            final String dbToken) {
+        Map<String, String> headers = new HashMap<String, String>() {
+            {
```

Comment:
can we refrain from using this pattern for maps? It creates an anonymous sub-class.

Reply:
For test I don't mind but this is SDK code.

Reply:
okie!... but it looks pretty neat..